src/Fixer/Alias/MbStrFunctionsFixer.php

```diff
@@ -58,7 +58,7 @@ public function __construct()
         $this->functions = array_filter(
             self::$functionsMap,
             static function (array $mapping): bool {
-                return \function_exists($mapping['alternativeName']) && (new \ReflectionFunction($mapping['alternativeName']))->isInternal();
+                return (new \ReflectionFunction($mapping['alternativeName']))->isInternal();
```

Comment:
Any reason for why the check with `function_exists` was removed? This seems to be a BC break on PHP 7.3.

`Uncaught ReflectionException: Function mb_str_split() does not exist in vendor/friendsofphp/php-cs-fixer/src/Fixer/Alias/MbStrFunctionsFixer.php:61`

Reply:
`symfony/polyfill-mbstring` was added as a dependency so this should not happen. How did you install PHP CS Fixer?

Reply:
Hello @julienfalque,

Well I'm removing polyfills when the extension is mandatory. I'm the only one who is doing this?

For example

```json
{
    "minimum-stability": "dev",
    "prefer-stable": true,
    "require": {
        "php": "^7.3",
        "ext-ctype": "*",
        "ext-mbstring": "*",
        "friendsofphp/php-cs-fixer": "^3.3"
    },
    "replace": {
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-mbstring": "*",
        "symfony/polyfill-php56": "*",
        "symfony/polyfill-php70": "*",
        "symfony/polyfill-php71": "*",
        "symfony/polyfill-php72": "*",
        "symfony/polyfill-php73": "*"
    }
}
```

I know it's a weird configuration, but I'm just curious about the reasoning.

Reply:
If you remove a requirement from the application there is nothing much we can do to help you, I mean these are in there for the reason that they are required. If you share how you installed the app like Julien asked we might be able to help, but otherwise not sure what we can do here.

Reply:
@SpacePossum I already shared my `composer.json`. That should be enough.

Reply:
> I'm the only one who is doing this?

No, I do that as well :)

Since you have `ext-mbstring`, this should work. Is the extension actually enabled?

Reply:
Yes, it's enabled. The issue is that I'm still on PHP 7.3, while [mb_str_split](https://www.php.net/manual/en/function.mb-str-split.php) is available on PHP 7 >= 7.4.0, PHP 8. That's why the `function_exists` check was relevant here.

Reply:
I see, then the check is still required indeed. If you don't mind, please submit a PR to fix that.

Reply:
how is the check still required, is it not in the polyfil?

Reply:
Without the polyfill, we would have to check whether the function exists even with `ext-mbstring` installed because the function was added in PHP 7.4. But the polyfill has a different behavior because it creates it regardless of the PHP version.

I'm actually not sure about this being a BC break. We can:
* consider using the polyfill is safe and we don't add the check back: people removing the polyfill do it at their own risk, but it means we rely on the polyfill's specific behavior, which one could considerer a bit risky
* or we consider the poolyfill's specific behavior a quirk and we need to add the check back because we still support some PHP versions before 7.4
* or we drop PHP 7.2 and 7.3 support, removing the problem entirely 😁️

I would go with last option but I'm fine with any.

Reply:
extention evolves over time, they add new functionality. we don't version the extension, but we can version the polyfill. i would expect nobody messing up with PHP CS Fixer requirements.
If you have extension in place, but old one (eg for PHP 7.1), then the polyfill will add only missing functionalities.

(drop of PHP 7.2/7.3 support is short term solution, the same concern would reappear later)

Reply:
So option 1: we keep current situation and don't add the check back?

Reply:
I'm having a hard time finding where it's documented (I'll updated if I find it), but this broke the default setup of Symfony 4.4 LTS running on php 7.3. 

Out of the box it has the described `composer.json` talked above which replaces the polyfill.